### DISCUSSION TITLE
New to termux and face a lot of problem, just would like to ask something knowledge and suggestion

### DISCUSSION THREAD

**Discussion by SamYokai** [termux/termux-packages#8402] [Q&A] [answered]
want to ask that currently I'm doing on android 5.1.1 platform, and I'm using termux 0.75, it can run, but I would like to ask that is it will face any problem if I run on my platform actually? Because I cannot run for termux-change-repo, manually also cannot, and apt update also will show 502 Bad Gateway for game and science, then during python install package also will face a lot of problem such as always red screen only and failed to install package. I know this platform is for phone so will different with PC but in python also a bit similar, but in termux Im face a lot of problem during python. So I just would like to know that is it my android platform that really not suggested to run termux?

**Comment by ghost** [comment, marked as answer]
> I'm doing on android 5.1.1

> but I would like to ask that is it will face any problem if I run on my platform actually?

Yes, Termux on Android 5.1.1 is not supported since 2020.01.01 (assume 2 years out of support). It doesn't receive bug fixes and new packages, maintainers will not look into its issues as well.

Currently Termux supports Android 7.0 as minimum. Lots of stuff has been added/fixed in the latest versions.

> Because I cannot run for termux-change-repo

Legacy repo compatible with Android 5.x is only https://termux.net. There no mirrors and no reason to have `termux-change-repo`.

> 502 Bad Gateway for game and science

```
apt purge science-repo game-repo
apt update
```

**Reply by SamYokai** [reply]
I see, ok understood already, thank you very much for your explanation

**Reply by SamYokai** [reply]
by the way, may I ask for another question? Inside Termux, how can I install opencv and scipy? Because I had try the its-pointless method but it only found in pkg list and doesn't show in pip list, and python also cannot get in since pip list haven't them. Or there have other method to import the module from pkg list into pip list?

**Reply by ghost** [reply]
`opencv` and `scipy` will be available if you **properly set up** the its-pointless repo subscription.

**Reply by SamYokai** [reply]
how to say properly set up? because I just follow the step :
curl -LO https://its-pointless.github.io/setup-pointless-repo.sh
bash setup-pointless-repo.sh
pkg install scipy
, but it doesn't show me in pip list. Is the step I did is wrong way?

***Updated***
I had tried again and this time Im using android 7, and I had success install scipy, but opencv still cannot although I install through pkg install, scipy also using pkg install and this time can get in pip list but opencv still haven't inside pip list.

One more, why if I pkg install python it will only python3.10 for me, and I had try install python 3.9.7, and once I run bash setup-pointless-repo.sh it will automatic upgrade my python from 3.9 to 3.10 but I would like to stay in 3.9 since all its-pointless need stay in python 3.9 more better.

Could you help to explain for me what Im doing wrong?

**Reply by ghost** [reply]
>  but it doesn't show me in pip list

`scipy` is installed as deb package, not as pip one.

>  I run bash setup-pointless-repo.sh it will automatic upgrade my python from 3.9 to 3.10 but I would like to stay in 3.9

Try `apt-mark hold python` to disable package upgrading.

**Reply by SamYokai** [reply]
> scipy is installed as deb package, not as pip one.

This I understand, now scipy is settle already i think, since I can found it in pip list. Just the problem is opencv, can pkg install but cannot found in pip list.

> Try apt-mark hold python to disable package upgrading.

ok, I will try it. Thank you very much.


Additional question again, actually how can we import pytorch (torch and torch vision) in Termux? Is it can do it or actually cannot?

**Reply by ghost** [reply]
Pytorch is not available for Termux at all.

**Reply by SamYokai** [reply]
owh, I see. Because I just want to test whether can use phone detect object or not only. Then also never my. Now I just would like to try can import cv2 and capture video and images only. Termux can do for this right?
